webnn/concat.https.any.js

```diff
@@ -0,0 +1,48 @@
+// META: title=test WebNN API concat operation
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: timeout=long
+
+'use strict';
+
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-concat
+
+const concatTests = () => {
+  const resources = loadTestData('/webnn/resources/test_data/concat.json');
+  const tests = resources.tests;
+  const inputsData = resources.inputsData;
+  const expectedData = resources.expectedData;
+  const targetTests = [];
+  for (const test of tests) {
+    const inputShapeValues = [];
+    const inputShapes = test.inputs.shape;
+    const inputDataSource = test.inputs.data;
+    const expectedDataSource = test.expected.data;
+    let position = 0;
+    for (const shape of inputShapes) {
+      const size = sizeOfShape(shape);
+      inputShapeValues.push({shape, data: inputsData[inputDataSource].slice(position, position + size)});
```

Comment:
This code is especial to `concat` op. To inputs of others operations, it's simply to set `data`. @huningxin